pkg/settings/setting_test.go

```diff
@@ -29,3 +30,27 @@ func TestIsRelease(t *testing.T) {
 		a.Equal(value, result, fmt.Sprintf("Expected value [%t] for key [%s]. Got value [%t]", value, key, result))
 	}
 }
+
+// TestSystemDefaultRegistryDefault tests that the default registry is either
+// "" (an empty string) or the build time value to ensure backward compatibility.
```

Comment:
What do you mean by build time ? Is something at build time setting a var in [InjectDefaults](https://github.com/rancher/rancher/blob/release/v2.6/scripts/build-server#L20) ? If so where is that. All this PR seems to do is add ability to set an ENV var.